src/util.ts

```diff
@@ -134,3 +93,9 @@ export const isPrimitive = (val: any): val is PrimitiveValue => {
     val === null
   );
 };
+
+
+export const singletonConstruct = <T extends Construct>(scope: Construct, id: string, create: (scope: Construct, id: string) => T): T => {
```

Comment:
Be sure to run prettier automatically in the ide or via cli before committing changes. I'm noticing a bunch of formatting changes and am wondering if prettier is running for you?

Reply:
Looks like it didn't format. Been having issues with my vscode where the tests and ts server are taking too much memory... likely prettier didn't get to run when I saved.

Maybe we need to add git hooks?

Reply:
https://github.com/sam-goodwin/functionless/pull/116